Packs/OktaASA/Integrations/OktaASA/OktaASA.py

```diff
@@ -0,0 +1,359 @@
+            create_time = arg_to_datetime(arg=event.get('timestamp'))
+            event['_time'] = create_time.strftime(DATE_FORMAT) if create_time else None
+
+
+'''COMMAND FUNCTIONS'''
+
+
+def test_module(client: OktaASAClient, team_name: str) -> str:
+    """
+    Tests API connectivity and authentication
+    When 'ok' is returned it indicates the integration works like it is supposed to and connection to the service is
+    successful.
+    Raises exceptions if something goes wrong.
+
+    Args:
+        client (OktaASAClient): OktaASAClient client to use.
+        team_name (str): the name of the team.
+
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+    try:
+        fetch_events_command(
+            client=client,
+            last_run={},
+            max_audit_events_per_fetch=1,
+            is_fetch_events=True,
+            team_name=team_name
+        )
+
+    except Exception as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+
+    return 'ok'
+
+
+def get_events_command(client: OktaASAClient, args: dict) -> tuple[List[Dict], CommandResults]:
+    """
+        Gets audit events from Audits Events endpoint.
+
+        Args:
+            self (OktaASAClient): Okta ASA Client.
+            args (dict): A dictionary containing the command arguments.
+
+        Returns:
+            List[Dict]: list of events.
+            CommandResults: command results containing Audits Events.
+    """
+
+    max_audit_events_per_fetch = arg_to_number(args.get('limit')) or 50
+    events, _, _ = client.search_events(
+        limit=max_audit_events_per_fetch,
+        offset=None,
+    )
+    hr = tableToMarkdown(name='Audits Events', t=events)
+    return events, CommandResults(readable_output=hr)
+
+
+def fetch_events_command(client: OktaASAClient, last_run: dict[str, str],
+                         team_name: str,
+                         max_audit_events_per_fetch: Optional[int],
+                         is_fetch_events: bool = False,
+                         ) -> tuple[dict[str, str], List[Dict]]:
+    """
+    Args:
+        client (OktaASAClient): OktaASAClient client to use.
+        last_run (dict): A dict with a key containing the latest event created time we got from last fetch.
+        max_audit_events_per_fetch (int): number of events per fetch.
+        is_fetch_events (bool): Whether the option to fetch events is selected.
+        team_name (str): The name of the team.
+    Returns:
+        dict: Next run dictionary containing the timestamp that will be used in ``last_run`` on the next fetch.
+        list: List of events that will be created in XSIAM.
+    """
+    if last_run and last_run.get("team_name") != team_name:
+        demisto.debug(f'{INTEGRATION_NAME}: Reset last run the name of the group has changed.')
+    if is_fetch_events:
```

Comment:
By default fetch_events_command should fetch events. Please remove this condition...